service/storage_service.cc

```diff
@@ -960,6 +960,16 @@ class topology_coordinator {
         return {};
     }
 
+    std::unordered_set<raft::server_id> parse_ignore_nodes(const node_to_work_on& node) {
+        if (node.rs->state == node_state::removing) {
+            return std::get<std::unordered_set<raft::server_id>>(node.req_param.value());
+        }
+        if (node.rs->state == node_state::replacing) {
+            return std::get<replace_param>(node.req_param.value()).ignored_ids;
+        }
+        return {};
```

Comment:
Maybe we should do `on_internal_error` here for now since we're not supposed to call `parse_ignore_nodes` in situations other than `removing`/`replacing`?